contracts/abstract/JBPayoutRedemptionPaymentTerminal3_1.sol

```diff
@@ -0,0 +1,1646 @@
+        netPayoutAmount,
+        decimals,
+        _projectId,
+        _group,
+        _split
+      );
+
+      // Trigger the allocator's `allocate` function.
+      // If this terminal's token is ETH, send it in msg.value.
+      try
+        _split.allocator.allocate{value: token == JBTokens.ETH ? netPayoutAmount : 0}(_data)
+      {} catch {
+        // Set the net payout amount to 0 to signal the reversion.
+        netPayoutAmount = 0;
+
+        // Add undistributed amount back to project's balance.
+        store.recordAddedBalanceFor(_projectId, _amount);
+      }
+
+      // Otherwise, if a project is specified, make a payment to it.
+    } else if (_split.projectId != 0) {
+      // Get a reference to the Juicebox terminal being used.
+      IJBPaymentTerminal _terminal = directory.primaryTerminalOf(_split.projectId, token);
+
+      // The project must have a terminal to send funds to.
+      if (_terminal == IJBPaymentTerminal(address(0))) revert TERMINAL_IN_SPLIT_ZERO_ADDRESS();
+
+      // If the terminal is set as feeless, this distribution is not eligible for a fee.
+      if (_feeDiscount == JBConstants.MAX_FEE_DISCOUNT || isFeelessAddress[address(_terminal)])
+        netPayoutAmount = _amount;
+        // This distribution is eligible for a fee since the funds are leaving this contract and the terminal isn't listed as feeless.
+      else {
+        unchecked {
+          netPayoutAmount = _amount - _feeAmount(_amount, fee, _feeDiscount);
+        }
+      }
+
+      // Trigger any inherited pre-transfer logic.
+      _beforeTransferTo(address(_terminal), netPayoutAmount);
+
+      // Send the projectId in the metadata.
+      bytes memory _projectMetadata = new bytes(32);
+      _projectMetadata = bytes(abi.encodePacked(_projectId));
+
+      // Add to balance if prefered.
+      if (_split.preferAddToBalance)
+        try
+          _terminal.addToBalanceOf{value: token == JBTokens.ETH ? netPayoutAmount : 0}(
+            _split.projectId,
+            netPayoutAmount,
+            token,
+            '',
+            _projectMetadata
+          )
+        {} catch {
+          // Set the net payout amount to 0 to signal the reversion.
+          netPayoutAmount = 0;
+
+          // Add undistributed amount back to project's balance.
+          store.recordAddedBalanceFor(_projectId, _amount);
+        }
+      else
+        try
+          _terminal.pay{value: token == JBTokens.ETH ? netPayoutAmount : 0}(
+            _split.projectId,
+            netPayoutAmount,
+            token,
+            _split.beneficiary != address(0) ? _split.beneficiary : msg.sender,
+            0,
+            _split.preferClaimed,
+            '',
+            _projectMetadata
+          )
+        {} catch {
+          // Set the net payout amount to 0 to signal the reversion.
+          netPayoutAmount = 0;
+
+          // Add undistributed amount back to project's balance.
+          store.recordAddedBalanceFor(_projectId, _amount);
```

Comment:
See [this comment](https://github.com/jbx-protocol/juice-contracts-v3/pull/17/files#r1103702332)